x-pack/plugin/sql/jdbc/src/test/java/org/elasticsearch/xpack/sql/jdbc/JdbcArrayTests.java

```diff
@@ -66,4 +75,90 @@ public void testArraySlicing() throws SQLException {
         sqle = expectThrows(SQLException.class, () -> array.getArray(Integer.MAX_VALUE + 1L, 9));
         assertEquals("Index value [2147483648] out of range [1, 2147483647]", sqle.getMessage());
     }
+
+    public void testSqlExceptionPastFree() throws SQLException {
+        Array array = new JdbcArray(jdbcTestConfiguration().timeZone(), baseType(EsType.LONG_ARRAY), emptyList());
+        array.free();
+
+        List<ThrowingRunnable> calls = asList(array::getBaseTypeName, array::getBaseType, array::getArray,
+            () -> array.getArray(emptyMap()), () -> array.getArray(1, 3), () -> array.getArray( 1, 2, emptyMap()),
+            array::getResultSet, () -> array.getResultSet(3, 5), () -> array.getResultSet(emptyMap()),
+            () -> array.getResultSet(1, 4, emptyMap())
+        );
+
+        for (ThrowingRunnable call : calls) {
+            SQLException sqle = expectThrows(SQLException.class, call);
+            assertEquals("Array has been freed already", sqle.getMessage());
+        }
+
+        array.free();
+    }
+
+    public void testNonEmptyMapRejected() throws SQLException {
+        Array array = new JdbcArray(jdbcTestConfiguration().timeZone(), baseType(EsType.LONG_ARRAY), emptyList());
+        Map<String, Class<?>> map = Map.of("foo", String.class);
+
+        List<ThrowingRunnable> calls = asList(() -> array.getArray(map), () -> array.getArray( 1, 2, map),
+            () -> array.getResultSet(map), () -> array.getResultSet(1, 4, map)
+        );
+
+        for (ThrowingRunnable call : calls) {
+            SQLException sqle = expectThrows(SQLException.class, call);
+            assertEquals("non-empty Map parameter not supported", sqle.getMessage());
+        }
+    }
+
+    public void testArrayGetAsResultSet() throws SQLException {
+        List<Integer> expected = randomList(1, 10, ESTestCase::randomInt);
+        Array array = new JdbcArray(jdbcTestConfiguration().timeZone(), baseType(EsType.INTEGER_ARRAY), expected);
+
+        ResultSet resultSet = array.getResultSet();
+        // meta
+        ResultSetMetaData meta = resultSet.getMetaData();
+        assertEquals(2, meta.getColumnCount());
+        assertEquals(Types.BIGINT, meta.getColumnType(1));
+        assertEquals("index", meta.getColumnName(1));
+        assertEquals(Types.INTEGER, meta.getColumnType(2));
+        assertEquals("value", meta.getColumnName(2));
+        // values
+        for (int i = 0; i < expected.size(); i++) {
+            assertTrue(resultSet.next());
+            assertEquals(Long.valueOf(i + 1), resultSet.getObject(1));
+            assertEquals(expected.get(i), resultSet.getObject(2));
+        }
+        assertFalse(resultSet.next());
+    }
+
+    public void testArrayGetAsSlicedResultSet() throws SQLException {
```

Comment:
Maybe add some randomness here instead of the constants.

Reply:
I could another one, but I'd like to keep this one as is, as it's very simple to follow how the slicing is done and what's expected. (and with some willingness from the gods of CI, not have to run the whole suite again)